lib/kafka/consumer.rb

```diff
@@ -203,7 +203,9 @@ def each_message(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automatica
       consumer_loop do
         batches = fetch_batches
 
-        batches.each do |batch|
+        # make sure any old batches, fetched prior to the completion of a consumer group sync,
+        # are only processed if the batches are from brokers for which this broker is still responsible.
+        batches.select { |batch| @group.subscribed_partitions[batch.topic].include?(batch.partition) }.each do |batch|
```

Comment:
Can you move this logic into `fetch_batches`?

Also, can you add a helper method to ConsumerGroup, e.g. `@group.assigned_to?(batch.topic, batch.partition)`?

Reply:
@dasch Yeah, I thought about putting it in `fetch_batches` in the first place, but couldn't decide where the responsibility should rest. I was thinking perhaps fetch should just fetch and not filter.  But happy to move it in there.

Will also create helper method.